### DISCUSSION TITLE
Dynamic routes

### DISCUSSION THREAD

**Discussion by theo-m** [trpc/trpc#1718] [Q&A]
Hey there, big fan of trpc! 

I was wondering if there was a plan to support dynamic routes, e.g. writing
```ts
trpc.router<MyContext>()
  .query("user/[id]", { resolve: ({ctx}) => { ctx.query.id }})
```

It kinda goes against the spirit of validating inputs, but it's a more standard way of exposing routes to the outside world too.

**Comment by KATT** [comment]
No, that won't happen, but you can do your own API handler that exposes that. See #1724 for inspiration on how you could implement an `/api/user/[id].ts`
